### PR TITLE
Addition: computed role mappings for html elements

### LINKED ISSUE #464: Add computedrole column for HTML-AAM
Based on the outcome of this ARIA issue…
- https://github.com/w3c/aria/issues/1887

HTML-AAM will likely need an update similar to:
- https://github.com/w3c/core-aam/pull/167

### PR DESCRIPTION
**Pull request by scottaohara**
resolves #464

The initial PR adds the computed role cell to all HTML elements.  Where possible, the cells have been filled in with 'use aria mappings', 'not mapped' or 'see comments' or '???'.  The ??? instances are those where there are host-language roles that need to be determined for elements which are important to accessibility but have no explicit ARIA role to neatly map to.

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/html-aam/pull/465.html" title="Last updated on Jun 6, 2023, 7:14 PM UTC (e6526e3)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/html-aam/465/105b6fc...e6526e3.html" title="Last updated on Jun 6, 2023, 7:14 PM UTC (e6526e3)">Diff</a>

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -871,6 +906,7 @@ <h3>HTML Element Role Mappings</h3>
                 <a data-cite="HTML">`dl`</a>
               </th>
               <td class="aria">No corresponding role</td>
+              <td class="role-computed"><div class="general">???</div></td>
```
Comment (cookiecrook) [inline index.html]: Isn't this `list`?

File: index.html [hunk 2]
```diff
@@ -2520,6 +2627,7 @@ <h3>HTML Element Role Mappings</h3>
                   <a href="https://html.spec.whatwg.org/#rb">`rb`</a>
                 </th>
                 <td class="aria">No corresponding role</td>
+                <td class="role-computed"><div class="general">???</div></td>
```
Comment (cookiecrook) [inline index.html]: Should there be a Ruby-AAM?

File: index.html [hunk 3]
```diff
@@ -2830,6 +2954,7 @@ <h3>HTML Element Role Mappings</h3>
             <tr tabindex="-1" id="el-style">
               <th><a data-cite="HTML">`style`</a></th>
               <td class="aria">No corresponding role</td>
+              <td class="role-computed"><div class="general">Not mapped</div></td>
```
Comment (cookiecrook) [inline index.html:2934]: One of my favorite old-school browser hacks was to set `display: block;` and `contenteditable` on a `<style>` block.

**Review by cookiecrook — APPROVED** [review]
With or without the `<dl>` comment, this is good to go to get the non-controversial column in. Would be nice to file a new issue or two to track what happens to audio/video and the various extra input types once we have consensus on https://github.com/w3c/aria/issues/529

File: index.html (hunk 1 above)
Comment (scottaohara) [reply index.html]: wasn't sure if we actually got consensus on that re the big 'associationlist' conversation

File: index.html (hunk 2 above)
Comment (scottaohara) [reply index.html]: probably!

File: index.html (hunk 3 above)
Comment (scottaohara) [reply index.html:2934]: why we need to solidify the minimum role proposal :)

File: index.html (hunk 1 above)
Comment (cookiecrook) [reply index.html]:
```suggestion
              <td class="role-computed"><div class="general">list</div> Editorial Note: This value may change upon resolution of <a href="https://github.com/w3c/aria/issues/1662">ARIA #1662</a>.</td>
```

**Comment by cookiecrook** [comment]
> @scottaohara wants to merge 4 commits into [gh-pages](https://github.com/w3c/html-aam) from [computed-role](https://github.com/w3c/html-aam/tree/computed-role)

Is merging into `gh-pages` expected for this project?

**Comment by scottaohara** [comment]
@spectranaut @cookiecrook finished removing the placeholder "???" i had for some computed roles, and added in a few notes relating to other issues where the computed roles may change (dl/dt/dd and the aria association list issue, header/footer)

**Pull request merged**